### DISCUSSION TITLE
React to database changes

### DISCUSSION THREAD

**Discussion by Soneji** [nocodb/nocodb#553] [Q&A]
Hi

I was wondering if NocoDB has the ability to react to database changes. I am using a MySQL backend with NocoDB but noticed that nothing happened when I added a new table from MySQL. 

The reason I'm after this feature is because I want to manage the MySQL db with NocoDB, and from herein use it for all the applications I build, but we have legacy applications that interface with mysql directly. 

Is this something that can be supported? does it refresh automatically and I didnt wait long enough? or is there a manual way to refresh nocodb?

Thanks in advance

**Comment by o1lab** [comment]
NocoDB works with a virtual schema on your database. Syncing changes with APIs running is a complex operation. So tread slowly here :) If changes are done outside to it - NocoDB can sync.

And is documented [here](https://docs.nocodb.com/setup-and-usages/sync-schema).

**Reply by Soneji** [reply]
Is there a way to sync it through the API?

**Reply by Soneji** [reply]
Usually all I'll be doing is creating tables, so there shouldnt be too much complexity ;p

**Reply by o1lab** [reply]
> Is there a way to sync it through the API?

You could theoretically call the API under the hood. But our metaapis arent documented and we are in the process of doing it.

**Reply by o1lab** [reply]
> Usually all I'll be doing is creating tables, so there shouldnt be too much complexity ;p

Should be good then.

However, things with relations it can get complicated.
